yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -616,7 +616,8 @@ private[spark] class Client(
     val appId = newAppResponse.getApplicationId
     val appStagingDir = getAppStagingDir(appId)
     val pySparkArchives =
-      if (sys.props.getOrElse("spark.yarn.isPython", "false").toBoolean) {
+      if (sys.props.getOrElse("spark.yarn.isPython", "false").toBoolean ||
+        sparkConf.getBoolean("spark.yarn.isPython", false)) {
```

Comment:
This could just be `sparkConf.getBoolean("spark.yarn.isPython", false)`, since `SparkConf` will read system properties.